lib/Models/WebProcessingServiceCatalogFunctionJob.ts

```diff
@@ -491,17 +496,21 @@ function formatOutputValue(title: string, value: string | undefined) {
   }, "");
 }
 
-/**
- * Transform old catalog definitions to match new schema.
- */
-function fixCatalogItemJson(json: any) {
-  const { isEnabled, ...fixedJson } = json;
-  if (json.type === "csv") {
-    const { data, tableStyle, ...fixedCsvJson } = fixedJson;
-    fixedCsvJson.csvString = fixedCsvJson.csvString || data;
-    return fixedCsvJson;
+async function tryConvertV7ToV8Catalog(
+  input: JsonObject
+): Promise<Record<string, unknown> | undefined> {
+  const { convertMember, convertCatalog } = await import("catalog-converter");
+  if (typeof input.type === "string") {
+    const { member, messages } = convertMember(input);
+    if (member === null || messages.length > 0)
+      throw new Error("Error converting v7 item to v8");
+    return member;
+  } else {
+    const { result, messages } = convertCatalog(input);
+    if (result === null || messages.length > 0)
+      throw new Error("Error converting v7 catalog to v8");
+    return result;
   }
```

Comment:
Use catalog-converter to properly convert terria json in WPS response. The response object could be a full catalog or a single catalog item.